### ISSUE TITLE
forEachFeatureAtPixel returns wrong layer if feature already selected

### ISSUE THREAD

**Issue by probins** [openlayers/ol3#3878]
Following situation used to work, but no longer does since removal of FeatureOverlay:

* 2 select interactions, one on mousemove, one on click, so user gets title on hover, and can click for more details. So when clicks, the feature is already selected. Click should return the selected feature, but event.selected is empty.

This seems to be because map.forEachFeatureAtPixel is returning the wrong layer: the overlay layer not the original vector layer, so it's failing the layer filter. I think the problem is around https://github.com/openlayers/ol3/blob/master/src/ol/renderer/maprenderer.js#L173 which I think should also be applying the layer filter so it only returns features on the original layers, not the overlay layer.

**Comment by ahocevar** [comment]
Thanks for the bug report. I'm working on this now.

**Comment by ahocevar** [comment]
@probins Can you please test #3883 to see if that pull request fixes the problem?

**Comment by akrherz** [comment]
@ahocevar For what it is worth, I tested this patch and it got me past the initial bug, but now I am hitting some other issue with attempting to keep track of a clicked feature and mouseover feature that I will have to research some more.

**Comment by probins** [comment]
that should fix it, but I'll take a look later today

**Comment by probins** [comment]
yes, that fixes my issue, thanks. Only situation that might not work with this is if you have unmanaged layers included in the layer filter - not sure whether that's a realistic use case, though. :-)

**Comment by ahocevar** [comment]
I added documentation to make clear that the forEachFeatureAtPixel layer filter does not filter unmanaged layers.

**Comment by probins** [comment]
This problem has returned.

Still the same situation of having 2 select interactions on the same layer/feature. Worked fine after this fix up to v3.11.2. Stops working with v3.12 (only just got around to upgrading, which is why I'm just reporting). There have been several versions of this, and I've got a bit confused as to what was changed where and why, but I think that https://github.com/openlayers/ol3/commit/279eae1dba5319c840c4e17429eb422c1b079355 (#4472) is the culprit.

Either interaction on its own works fine but, with both together, the 2nd one triggers the interaction but evt.selected is empty. There is no filter function, only a layer option restricting selection to the vector layers.
